src/main/java/org/isf/menu/service/UserMenuItemIoOperationRepository.java

```diff
@@ -36,7 +36,7 @@ public interface UserMenuItemIoOperationRepository extends JpaRepository<UserMen
 			"menuItem.mySubmenu, menuItem.myClass, menuItem.isASubMenu, menuItem.position, groupMenu.active " +
 			"from UserMenuItem menuItem, GroupMenu groupMenu, UserGroup  userGroup, User user " +
 			"where (user.userName=:userId) " +
-			"and (user.userGroupName=userGroup.code) " +
+			"and (user.userGroupName.code=userGroup.code) " +
```

Comment:
wondering how it was working before :/

Reply:
Same question I had; there are others like this too (including some of the tests worked when they shouldn't have)

Reply:
No; those lines are correct as they are joining on `string`s.   Line 39 was trying to join on the variable `userGroupName which is an object  of type `UserGroup` with a `string` code value in the `userGroup` object.   Not sure how it was ever working but then the newer Hibernate and Spring JPA seem to be more strict.